### PR TITLE
Friendly URLs

### LINKED ISSUE #10: Nice SEO friendly URLs
because it looks nicer and google might like it better: 

- [ ] Use category name in the URLs
- [ ] Add entry title as part of the URL

### PR DESCRIPTION
**Pull request by Plsr**
* Add [FriendlyId](https://github.com/norman/friendly_id) to help dealing with slugs
* Use slugs for categories and posts, leading to nicer urls
  * post slugs are limited to 80 characters, so in case we have a post with a long and duplicate title, we still have a reasonable length for the url (80 characters + `https://decent-organisations.tld/posts/` + 36 characters UUID generated by FriendlyId in case of duplicate slugs)

**NOTE**: After the migrations have run, all existing posts and categories need to be touched and saved once to populate the slugs. Is it sufficient to run `Category.find_each(&:save)` and `Post.find_each(&:save)` in the console or do we need a rake script (or something else, I'm honestly not too sure what the best approach would be)?

Closes #10

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class Category < ApplicationRecord
+  extend FriendlyId
+  friendly_id :name, use: :slugged
+
   has_and_belongs_to_many :posts
 
   validates :name, presence: true, length: { maximum: 255 }, uniqueness: true
   validates :description, length: { maximum: 1000 }
+  validates :slug, presence: true
```
Comment (bumi) [inline app/models/category.rb:9]: isn't friendly_id already taking care of that?

File: db/migrate/20180222131221_create_friendly_id_slugs.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CreateFriendlyIdSlugs < ActiveRecord::Migration[5.1]
+  def change
+    create_table :friendly_id_slugs do |t|
+      t.string   :slug,           :null => false
+      t.integer  :sluggable_id,   :null => false
+      t.string   :sluggable_type, :limit => 50
+      t.string   :scope
+      t.datetime :created_at
+    end
```
Comment (bumi) [inline db/migrate/20180222131221_create_friendly_id_slugs.rb:9]: oha, what is friendly_id doing with this table?

File: app/models/category.rb (hunk 1 above)
Comment (Plsr) [reply app/models/category.rb:9]: In their guide, they also validate the presence of the slug, see http://norman.github.io/friendly_id/file.Guide.html#Example_Setup

Not sure how up to date these guides are, though.

Comment (bumi) [reply app/models/category.rb:9]: ok, thx. 
und schadet bestimmt eh nicht. :)

File: db/migrate/20180222131221_create_friendly_id_slugs.rb (hunk 2 above)
Comment (Plsr) [reply db/migrate/20180222131221_create_friendly_id_slugs.rb:9]: Looks like it is used for keeping a history of a models' slugs https://stackoverflow.com/a/22882365/4181679

Avoiding broken urls sounds like a good idea to me, so maybe I.. should actually use `:history` and not just create a table that will not be used :D

**Pull request merged**